src/ray/gcs/gcs_server/test/gcs_server_rpc_test.cc

```diff
@@ -127,6 +127,47 @@ class GcsServerTest : public ::testing::Test {
     AsyncCall(call_function, timeout_ms_);
   }
 
+  void TestRegisterNodeInfo(const rpc::RegisterNodeInfoRequest &request) {
+    auto call_function = [this, request](std::promise<bool> &promise) {
+      client_->RegisterNodeInfo(
+          request,
+          [&promise](const Status &status, const rpc::RegisterNodeInfoReply &reply) {
+            RAY_CHECK_OK(status);
+            promise.set_value(true);
+          });
+    };
+    AsyncCall(call_function, timeout_ms_);
+  }
+
+  void TestUnregisterNodeInfo(const rpc::UnregisterNodeInfoRequest &request) {
+    auto call_function = [this, request](std::promise<bool> &promise) {
+      client_->UnregisterNodeInfo(
+          request,
+          [&promise](const Status &status, const rpc::UnregisterNodeInfoReply &reply) {
+            RAY_CHECK_OK(status);
+            promise.set_value(true);
+          });
+    };
+    AsyncCall(call_function, timeout_ms_);
+  }
+
+  void GetAllNodesInfo(std::vector<rpc::GcsNodeInfo> &node_infos) {
+    node_infos.clear();
+    rpc::GetAllNodesInfoRequest request;
+    auto call_function = [this, request, &node_infos](std::promise<bool> &promise) {
+      client_->GetAllNodesInfo(
+          request, [&promise, &node_infos](const Status &status,
+                                           const rpc::GetAllNodesInfoReply &reply) {
+            RAY_CHECK_OK(status);
+            promise.set_value(true);
+            for (int index = 0; index < reply.node_infos_size(); ++index) {
+              node_infos.push_back(reply.node_infos(index));
+            }
+          });
+    };
+    AsyncCall(call_function, timeout_ms_);
+  }
+
   void AsyncCall(const CallFunction &function, uint64_t timeout_ms) {
```

Comment:
It is more better to define `using CallFunction  = std::function<std::future<bool>()>;`, so that we need not to declare  `std::promise<bool>` inner each test.

Reply:
Just like:
```
auto future = function();
auto status = future.wait_for(std::chrono::milliseconds(timeout_ms));
ASSERT_EQ(status, std::future_status::ready);
```

Reply:
Besides, the `AsyncCall` is confusing. Maybe better to redefine as `bool IsFutureReady(const CallFunction &function, uint64_t timeout_ms)` and do not use ASSERT_EQ inside it. If you want to check something, just use `ASSERT_EQ(IsFutureReady(future, 200));`

Reply:
fixed